### PR TITLE
Add concurrency limits on sqs runner

### PR DESCRIPTION
**Pull request by johndagostino**
Add concurrency limits on sqs runner.

This just uses bluebird to prevent overloading the consumer

### CODE REVIEW

File: src/runner/sqs.ts [hunk 1]
```diff
@@ -117,12 +128,13 @@ class SqsRunner extends BaseRunner implements IRunner {
           this.registry.events.emit('runner_failure', topic, ex, params);
         }
         this.pool.release(resource);
-      })
+      },
+      { concurrency: this.concurrency }
```
Comment (rrsmith) [inline src/runner/sqs.ts:132]: 🎉

File: src/runner/sqs.ts [hunk 2]
```diff
@@ -147,24 +159,26 @@ class SqsRunner extends BaseRunner implements IRunner {
...
+            VisibilityTimeout: config.visibilityTimeout,
+            WaitTimeSeconds: config.waitTimeSeconds,
           };
           await this.dequeue(topic, params);
         } else {
           this.logger.error(`Queue URL ${topic} not found`);
         }
-      })
+      },
+      { concurrency: this.concurrency }
```
Comment (rrsmith) [inline src/runner/sqs.ts:181]: I had been going to default this to subscriptions.length for 1 at a time message per subscription processing by default. Not sure if it'd add a huge amount at this point though

File: src/config/sqs.ts [hunk 3]
```diff
@@ -16,21 +16,7 @@ const sqs = (config: Configuration) => {
...
-  const listQueues = instance.listQueues.bind(instance);
-  const dummy = {
-    createQueueAsync: (...args) => createQueue(...args).promise(),
-    sendMessageAsync: (...args) => sendMessage(...args).promise(),
-    receiveMessageAsync: (...args) => receiveMessage(...args).promise(),
-    getQueueUrlAsync: (...args) => getQueueUrl(...args).promise(),
-    deleteMessageAsync: (...args) => deleteMessage(...args).promise(),
-    listQueuesAsync: (...args) => listQueues(...args).promise(),
-  };
-  return Object.assign(instance, dummy);
```
Comment (arshu16) [inline src/config/sqs.ts:33]: 🙌

File: src/runner/sqs.ts [hunk 4]
```diff
@@ -203,7 +218,7 @@ class SqsRunner extends BaseRunner implements IRunner {
         MessageRetentionPeriod: messageRetentionPeriod,
       },
     };
-    return this.sqs.createQueueAsync(params);
+    await this.sqs.createQueue(params).promise();
```
Comment (rrsmith) [inline src/runner/sqs.ts:221]: Any particular reason to move away from just returning?

**Review by arshu16 — APPROVED** [review]

**Review by rrsmith — APPROVED** [review]
LGTM

File: src/runner/sqs.ts (hunk 4 above)
Comment (johndagostino) [reply src/runner/sqs.ts:221]: @rrsmith the type on the base runner is void, so this was triggering the type check

File: src/runner/sqs.ts (hunk 2 above)
Comment (johndagostino) [reply src/runner/sqs.ts:181]: @rrsmith yeah I want to make it smarter, potentially depending on WorkerConfig for the concurrency options.

Let me revisit in the next PR.

File: src/runner/sqs.ts (hunk 4 above)
Comment (rrsmith) [reply src/runner/sqs.ts:221]: Ahhh, make sense

File: src/runner/sqs.ts (hunk 2 above)
Comment (rrsmith) [reply src/runner/sqs.ts:181]: Sounds good 👍

**Pull request merged**